test/Conversion/SCFToCalyx/convert_controlflow.mlir

```diff
@@ -641,3 +641,122 @@ module {
+// CHECK:               calyx.assign %[[VAL_43]] = %[[VAL_68]] ? %[[VAL_6]] : i1
+// CHECK:               calyx.group_done %[[VAL_40]] : i1
+// CHECK:             }
+// CHECK:             calyx.group @bb0_1 {
+// CHECK:               calyx.assign %[[VAL_26]] = %[[VAL_39]] : i32
+// CHECK:               calyx.assign %[[VAL_27]] = %[[VAL_39]] : i32
+// CHECK:               calyx.assign %[[VAL_29]] = %[[VAL_28]] : i1
+// CHECK:               calyx.assign %[[VAL_30]] = %[[VAL_6]] : i1
+// CHECK:               calyx.group_done %[[VAL_34]] : i1
+// CHECK:             }
+// CHECK:             calyx.group @bb0_2 {
+// CHECK:               calyx.assign %[[VAL_10]] = %[[VAL_9]] : i32
+// CHECK:               calyx.assign %[[VAL_48]] = %[[VAL_11]] : i7
+// CHECK:               calyx.assign %[[VAL_51]] = %[[VAL_6]] : i1
+// CHECK:               calyx.assign %[[VAL_52]] = %[[VAL_7]] : i1
+// CHECK:               calyx.assign %[[VAL_20]] = %[[VAL_54]] : i32
+// CHECK:               calyx.assign %[[VAL_21]] = %[[VAL_55]] : i1
+// CHECK:               calyx.group_done %[[VAL_25]] : i1
+// CHECK:             }
+// CHECK:             calyx.group @bb0_3 {
+// CHECK:               calyx.assign %[[VAL_12]] = %[[VAL_8]] : i32
+// CHECK:               calyx.assign %[[VAL_48]] = %[[VAL_13]] : i7
+// CHECK:               calyx.assign %[[VAL_51]] = %[[VAL_6]] : i1
+// CHECK:               calyx.assign %[[VAL_52]] = %[[VAL_7]] : i1
+// CHECK:               calyx.assign %[[VAL_14]] = %[[VAL_54]] : i32
+// CHECK:               calyx.assign %[[VAL_15]] = %[[VAL_55]] : i1
+// CHECK:               calyx.group_done %[[VAL_19]] : i1
+// CHECK:             }
+// CHECK:             calyx.group @ret_assign_0 {
+// CHECK:               calyx.assign %[[VAL_62]] = %[[VAL_60]] : i32
+// CHECK:               calyx.assign %[[VAL_63]] = %[[VAL_6]] : i1
+// CHECK:               calyx.group_done %[[VAL_67]] : i1
+// CHECK:             }
+// CHECK:           }
+// CHECK:           calyx.control {
+// CHECK:             calyx.seq {
+// CHECK:               calyx.enable @bb0_0
+// CHECK:               calyx.enable @bb0_1
+// CHECK:               calyx.if %[[VAL_33]] {
+// CHECK:                 calyx.seq {
+// CHECK:                   calyx.enable @bb0_2
+// CHECK:                   calyx.enable @then_br_0
+// CHECK:                 }
+// CHECK:               } else {
+// CHECK:                 calyx.seq {
+// CHECK:                   calyx.enable @bb0_3
+// CHECK:                   calyx.enable @else_br_0
+// CHECK:                 }
+// CHECK:               }
+// CHECK:               calyx.enable @ret_assign_0
+// CHECK:             }
+// CHECK:           }
+// CHECK:         } {toplevel}
+  func.func @main(%arg0 : i32) -> i32 {
+    %1 = memref.alloc() : memref<120xi32>
+    %idx_one = arith.constant 1 : index
+    %two = arith.constant 2: i32
+    %rem = arith.remui %arg0, %two : i32
+    %cond = arith.cmpi eq, %arg0, %rem : i32
+
+    %res = scf.if %cond -> i32 {
```

Comment:
Nit: could we minimize what you're trying to test even further, e.g., do we need the `else` case, mem alloc, ...? It seems the critical piece is an if statement that uses the output of a sequential operation.